File: src/java/com/portalpostal/dao/handler/VeiculoHandler.java
package com.portalpostal.dao.handler;

import com.portalpostal.model.dd.TipoCombustivelVeiculo;
import com.portalpostal.model.dd.TipoSituacaoVeiculo;
import com.portalpostal.model.dd.TipoStatusVeiculo;
import com.portalpostal.model.dd.TipoVeiculo;
import com.portalpostal.model.Veiculo;
import java.sql.ResultSet;
import java.sql.SQLException;
import org.sql2o.ResultSetHandler;

public class VeiculoHandler extends GenericHandler implements ResultSetHandler<Veiculo> {
    
    public VeiculoHandler() { 
        super("veiculo");
    }
    
    public VeiculoHandler(String table) {
        super(table);
    }

    @Override
    public Veiculo handle(ResultSet result) throws SQLException {        
        Veiculo veiculo = new Veiculo();
        veiculo.setIdVeiculo(getInt(result, "idVeiculo"));
        veiculo.setTipo(TipoVeiculo.values()[getInt(result, "tipo")]);  
        veiculo.setIdMarca(getInt(result, "idMarca"));
        veiculo.setMarca(getString(result, "marca"));
        veiculo.setIdModelo(getInt(result, "idModelo"));
        veiculo.setModelo(getString(result, "modelo")); 
        veiculo.setIdVersao(getString(result, "idVersao"));
        veiculo.setVersao(getString(result, "versao"));  
        veiculo.setPlaca(getString(result, "placa"));
        veiculo.setAnoModelo(getInt(result, "anoModelo"));
        veiculo.setChassis(getString(result, "chassis"));
        veiculo.setRenavam(getString(result, "renavam"));
        veiculo.setQuilometragem(getInt(result, "quilometragem"));
        veiculo.setCombustivel(TipoCombustivelVeiculo.values()[getInt(result, "combustivel")]);
        veiculo.setStatus(TipoStatusVeiculo.values()[getInt(result, "status")]);
        veiculo.setSituacao(TipoSituacaoVeiculo.values()[getInt(result, "situacao")]);
        veiculo.setDataCadastro(getDate(result, "dataCadastro"));
        return veiculo;
    }
    
}

File: web/app/js/handler/veiculoHandler.js
'use strict';

app.factory('VeiculoHandler', function() {
    
    var handleList = function(veiculoList) { 
        return _.map(veiculoList, function(veiculo) {
            return handle(veiculo);
        });
    };
    
    var handle = function(veiculo) {  
        var veiculoHandle = {};
        veiculoHandle.idVeiculo = getId(veiculo); 
        veiculoHandle.tipo = getTipo(veiculo);
        veiculoHandle.idMarca = getCodigoMarca(veiculo);
        veiculoHandle.marca = getMarca(veiculo);
        veiculoHandle.idModelo = getCodigoModelo(veiculo);
        veiculoHandle.modelo = getModelo(veiculo);
        veiculoHandle.idVersao = getCodigoVersao(veiculo);
        veiculoHandle.versao = getVersao(veiculo);
        veiculoHandle.placa = getPlaca(veiculo);
        veiculoHandle.anoModelo = getAnoModelo(veiculo);
        veiculoHandle.chassis = getChassis(veiculo);
        veiculoHandle.renavam = getRenavam(veiculo);
        veiculoHandle.quilometragem = getQuilometragem(veiculo);
        veiculoHandle.dataCadastro = getDataCadastro(veiculo);
        return veiculoHandle;
    };
    
    var getId = function(veiculo) {
        return veiculo.idVeiculoCombustivel || null;
    };
    
    var getTipo = function(veiculo) {
        if(!veiculo.tipo) return veiculo.tipo;
        return veiculo.tipo.id; 
    };
    
    var getCodigoMarca = function(veiculo) {
        if(!veiculo.marca) return veiculo.marca;
        return veiculo.marca.id; 
    };
    
    var getMarca = function(veiculo) {
        if(!veiculo.marca) return veiculo.marca;
        return veiculo.marca.name; 
    };
    
    var getCodigoModelo = function(veiculo) {
        if(!veiculo.modelo) return veiculo.modelo;
        return veiculo.modelo.id; 
    };
    
    var getModelo = function(veiculo) {
        if(!veiculo.modelo) return veiculo.modelo;
        return veiculo.modelo.name; 
    };
    
    var getCodigoVersao = function(veiculo) {
        if(!veiculo.versao) return veiculo.versao;
        return veiculo.versao.id; 
    };
    
    var getVersao = function(veiculo) {
        if(!veiculo.versao) return veiculo.versao;
        return veiculo.versao.name; 
    };
    
    var getPlaca = function(veiculo) {
        return veiculo.placa || null; 
    };
    
    var getAnoModelo = function(veiculo) {
        return veiculo.anoModelo || null; 
    };
    
    var getChassis = function(veiculo) {
        return veiculo.chassis || null; 
    };
    
    var getRenavam = function(veiculo) {
        return veiculo.renavam || null; 
    };
    
    var getQuilometragem = function(veiculo) {
        return veiculo.quilometragem || null; 
    };
    
    var getDataCadastro = function(veiculo) {
        return veiculo.dataCadastro || null; 
    };

    return {
        handle: handle,
        handleList: handleList
    }

});